moodlecli/utils.py

```diff
@@ -343,3 +298,78 @@ def _attempts_data_is_stale(attempts, user_id, quiz_id, new_attempt_ts):
             new_grades["attempts"][user_id][quiz_id] = tmp_attempts
 
     return new_grades
+
+
+def course_creation(
```

Comment:
A couple of minor comments:

* The function name here is inconsistent with the file in that the rest are `{verb}_{noun}`. 
* It might be nice to use a function docstring to describe this for our future selves in terms of how it relates to `setup_duplicate_course` and why it's factored out